src/librustc_mir/transform/const_prop.rs

```diff
@@ -527,11 +527,11 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
         right: &Operand<'tcx>,
         source_info: SourceInfo,
     ) -> Option<()> {
-        let r =
```

Comment:
why are any of the changes in this function necessary?

Reply:
In `const_prop` we call `check_binary_op(...)?`, because `check_binary_op` requires both arguments to be defined, we fail in cases like `a | true`. I just lifted that restriction so that `check_binary_op` still does the same verification when the operation is something that could overflow but otherwise doesn't pre-emptively abort constant propagation.

Reply:
Ah makes sense, thanks!